spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -256,11 +256,25 @@ public void setEncoding(String encoding) {
 	 * A queue for replies; if not provided, a temporary exclusive, auto-delete queue will
 	 * be used for each reply, unless RabbitMQ supports 'amq.rabbitmq.reply-to' - see
 	 * http://www.rabbitmq.com/direct-reply-to.html
-	 *
+	 * @deprecated - use #setReplyAddress(String replyAddress)
 	 * @param replyQueue the replyQueue to set
 	 */
+	@Deprecated
 	public void setReplyQueue(Queue replyQueue) {
-		this.replyQueue = replyQueue;
+		setReplyAddress(replyQueue.getName());
+	}
+
+	/**
+	 * An address for replies; if not provided, a temporary exclusive, auto-delete queue will
+	 * be used for each reply, unless RabbitMQ supports 'amq.rabbitmq.reply-to' - see
+	 * http://www.rabbitmq.com/direct-reply-to.html
+	 * <p>The address can be a simple queue name (in which case the reply will be routed via the default
+	 * exchange), or with the form {@code exchange/routingKey} to route the reply using an explicit
+	 * exchange and routing key.
+	 * @param replyQueue the replyQueue to set
```

Comment:
s/replyQueue/replyAddress
and `@since 1.5`

Can be addressed during merge if I won't find anything else